software/controller/lib/core/alarm.cpp

```diff
@@ -0,0 +1,102 @@
+#include <stdint.h>
+#include "vars.h"
+#include "debug.h"
+#include "hal.h"
+#include "alarm.h"
+#include "comms.h"
+#include "commfail_alarm.h"
+
+// MAX_CONTROLLER_ALARMS 
+#define MAX_CONTROLLER_ALARMS 2
+
+// structure definition for MAX_CONTROLLER_ALARMS 
+alarm_cb_t alarms[MAX_CONTROLLER_ALARMS];
```

Comment:
Quite a bit of stuff in the next 15 lines or so feels very dangerous to me. We should not have all these global statically initialized variables (other than `constexpr` which is warranted). They should probably be encapsulated either in some sort of singleton or in an explicit instance of initialized in the main loop.